### DISCUSSION TITLE
Consumer does not use async with eventlet

### DISCUSSION THREAD

**Discussion by gonzalesMK** [celery/celery#7467] [Q&A]
Hey! I'm trying to use a consumer to process custom messages in a async fashion, but it happens that even using the eventlet option this code is executed in one thread ( I mean, this consumer finishes one message before getting to next, and it does not matter how many concurrency I set).

Could someone give me a glimpse of why this is happening ?
```
from celery import bootsteps
from kombu import Consumer, Exchange, Queue
class MyConsumerStep(bootsteps.ConsumerStep):
    def get_consumers(self, channel):
        return [
            Consumer(
                channel,
                queues=[Queue("feeds", Exchange("teste", type="topic"), routing_key="abc")],
                callbacks=[self.handle_message],
                accept=["json"],
            )
        ]

    def handle_message(self, body, message):
        print("Received message: {0!r}".format(body))
        time.sleep(5)
        message.ack()
```
(This code is the simple example given in the documentation )

**Comment by akarnavasco** [comment]
This was discussed in #6702.  This happens because the consumer is run on the MainProcess regardless of the pool implementation (same behavior happens with threads and processes implementation too). Because of that no processing should take place in the `handle_message` function but the message should be forwarded to a WorkerProcess. I'm in the dark as to how this is done without sending a new message through the broker and I would like some guidance because I face the same issue.
I have opened an issue in [stackoverflow](https://stackoverflow.com/questions/76798075/running-celery-tasks-using-custom-consumerstep).
